verl/trainer/ppo/ray_trainer.py

```diff
@@ -993,19 +1012,23 @@ def fit(self):
                     batch_keys=batch_keys_to_pop,
                     non_tensor_batch_keys=non_tensor_batch_keys_to_pop,
                 )
+                nvtx.end_range(nvtx_batch)
 
                 is_last_step = self.global_steps >= self.total_training_steps
 
                 with _timer("step", timing_raw):
                     # generate a batch
+                    nvtx_gen = nvtx.start_range(message="gen", color="green")
```

Comment:
what about adding the nvtx code to _timer instead?

Reply:
done. Indeed a better idea.